### PR TITLE
Fix atmosphere support

### PR DESCRIPTION
**Pull request by WalkerOfBacon**
figured it out now

### CODE REVIEW

**Comment by LiveFireExercise** [comment]
the problem is that this adonis creates an atmosphere, which may modify the game's lighting in unknown ways

**Comment by WalkerOfBacon** [comment]
> the problem is that this adonis creates an atmosphere, which may modify the game's lighting in unknown ways

In what ways?

**Comment by LiveFireExercise** [comment]
> > the problem is that this adonis creates an atmosphere, which may modify the game's lighting in unknown ways
> 
> In what ways?

well atmospheres spawn with some default properties. so i think a better idea would be to only create the atmosphere variables if an atmosphere is present

**Comment by WalkerOfBacon** [comment]
> well atmospheres spawn with some default properties. so i think a better idea would be to only create the atmosphere variables if an atmosphere is present

I think I did that already.

Adonis will only spawn in a new atmosphere if there isnt an existing one on spawn

**Comment by WalkerOfBacon** [comment]
It wont spawn it if an atmosphere is already in the game

**Comment by LiveFireExercise** [comment]
> It wont spawn it if an atmosphere is already in the game

thats the thing, adonis shouldnt spawn one at all.

**Comment by WalkerOfBacon** [comment]
> thats the thing, adonis shouldnt spawn one at all.

So then how would this work without the error?

**Comment by LiveFireExercise** [comment]
Simply only assign the variables a value if an atmosphere is present. either that or wrap it in something

**Comment by WalkerOfBacon** [comment]
Ill try and do that

**Comment by WalkerOfBacon** [comment]
@LiveFireExercise check it now

File: MainModule/Server/Commands/Moderators.lua [hunk 1]
```diff
@@ -2409,16 +2409,30 @@ return function(Vargs, env)
...
+				if service.Lighting:FindFirstChildWhichIsA("Atmosphere") then
+					Variables.OriginalAtmosphereSettings = {
+						Name = service.Lighting:FindFirstChildWhichIsA("Atmosphere").Name,
+						Density = service.Lighting:FindFirstChildWhichIsA("Atmosphere").Density,
+						Offset = service.Lighting:FindFirstChildWhichIsA("Atmosphere").Offset,
+						Color = service.Lighting:FindFirstChildWhichIsA("Atmosphere").Color,
+						Decay = service.Lighting:FindFirstChildWhichIsA("Atmosphere").Decay,
+						Glare = service.Lighting:FindFirstChildWhichIsA("Atmosphere").Glare,
+						Haze = service.Lighting:FindFirstChildWhichIsA("Atmosphere").Haze,
+					}
```
Comment (ccuser44) [inline MainModule/Server/Commands/Moderators.lua:2421]: You should add code to Process.lua which would do this on startup

**Review by ccuser44 — CHANGES_REQUESTED** [review]

**Comment by WalkerOfBacon** [comment]
intermission

**Comment by WalkerOfBacon** [comment]
I wonder when this will get merged

**Comment by LiveFireExercise** [comment]
Well now the variables are only being created if atmosphere exists

**Comment by WalkerOfBacon** [comment]
> Well now the variables are only being created if atmosphere exists

Is there a problem with that?

**Comment by LiveFireExercise** [comment]
If an atmosphere command is run then adonis will try and call variables that do not exist

**Comment by WalkerOfBacon** [comment]
> If an atmosphere command is run then adonis will try and call variables that do not exist

But if the variables dont exist then it would say that the atmosphere doesn't exist

**Comment by WalkerOfBacon** [comment]
I specifically listed that in the command

**Comment by WalkerOfBacon** [comment]
The command will only work if there is an atmosphere in Lighting

Otherwise it wont

**Comment by WalkerOfBacon** [comment]
It's not like its creating a new atmosphere

**Comment by LiveFireExercise** [comment]
And does it work

**Comment by WalkerOfBacon** [comment]
> And does it work

Considering its your command you made, im not sure
I just went off by what you made

**Comment by WalkerOfBacon** [comment]
ok

**Pull request merged**